src/ssh/azext_ssh/rdp_utils.py

```diff
@@ -0,0 +1,204 @@
+
+    # without Shell=True the print gets messed up once the subprocess dies.
+    # Is there a way to fix it? There are advantages to not use shell=True.
+    ssh_sub = subprocess.Popen(command, shell=True, stderr=subprocess.PIPE, env=env, encoding='utf-8')
+    return ssh_sub, print_ssh_logs
+
+
+def wait_for_ssh_connection(ssh_sub, print_ssh_logs):
+    log_list = []
+    ssh_sucess = False
+    while True:
+        next_line = ssh_sub.stderr.readline()
+        if print_ssh_logs:
+            print(next_line, end='')
+        else:
+            log_list.append(next_line)
+        if "debug1: Entering interactive session." in next_line:
+            logger.debug("SSH Connection estalished succesfully.")
+            ssh_sucess = True
+            break
+        if ssh_sub.poll() is not None:
+            logger.debug("SSH Connection failed.")
+            ssh_sucess = False
+            break
+    return ssh_sucess, log_list
+
+
+def terminate_ssh(ssh_process, log_list, print_ssh_logs):
+    if ssh_process:
+        terminated = False
+        if ssh_process.poll() is None:
+            kill_process(ssh_process.pid)
+            t0 = time.time()
+            while ssh_process.poll() is None and (time.time() - t0) < const.RDP_TERMINATE_SSH_WAIT_TIME_IN_SECONDS:
+                time.sleep(1)
+            terminated = True
+        print_error_messages_from_log(log_list, print_ssh_logs, ssh_process, terminated)
+
+
+def print_error_messages_from_log(log_list, print_ssh_logs, ssh_process, terminated):
+    # Read the remaining log messages since the connection was established.
+    next_line = ssh_process.stderr.readline()
+    while next_line:
+        if print_ssh_logs:
+            print(next_line, end='')
+        else:
+            log_list.append(next_line)
+        next_line = ssh_process.stderr.readline()
+
+    # If ssh process was not forced to terminate, print potential error messages.
+    if ssh_process.returncode != 0 and not print_ssh_logs and not terminated:
+        for line in log_list:
+            if "debug1:" not in line and line != '':
+                print(str(line))
+
+
+def _get_rdp_path(rdp_command="mstsc"):
+    rdp_path = rdp_command
+    if platform.system() == 'Windows':
+        arch_data = platform.architecture()
+        sys_path = 'System32'
+        system_root = os.environ['SystemRoot']
+        system32_path = os.path.join(system_root, sys_path)
+        rdp_path = os.path.join(system32_path, (rdp_command + ".exe"))
+        logger.debug("Platform architecture: %s", str(arch_data))
+        logger.debug("System Root: %s", system_root)
+        logger.debug("Attempting to run rdp from path %s", rdp_path)
+
+        if not os.path.isfile(rdp_path):
+            raise azclierror.BadRequestError("Could not find " + rdp_command + ".exe. Is the rdp client installed?")
+    else:
+        raise azclierror.BadRequestError("Platform is not supported for this command. Supported platforms: Windows")
```

Comment:
Could we potentially make this cross-platform by call the address using the `rdp://` protocol handler instead of calling the `.exe` directly?

Reply:
@zhoxing-ms if we cannot make this cross platform, is there some way to only show the parameter on Windows machines and not on Mac/Linux?

Reply:
@jiasli  Do you have any suggestions on this question?